acpi/src/fadt.rs

```diff
@@ -128,3 +131,27 @@ impl Fadt {
         }
     }
 }
+
+/// Information about the ACPI Power Management Timer (ACPI PM Timer).
+pub struct PmTimer {
+    /// An I/O space address to the register block of ACPI PM Timer.
+    pub io_base: u32,
+    /// This field is true if the hardware supports 32-bit timer, and false if the hardware
+    /// supports 24-bit timer.
+    pub supports_32bit: bool,
+}
+impl PmTimer {
+    /// Creates a new instance of `PmTimer`.
+    pub fn new<H>(tables: &AcpiTables<H>) -> Result<PmTimer, AcpiError>
+    where
+        H: AcpiHandler,
+    {
+        let fadt = unsafe {
+            tables
+                .get_sdt::<Fadt>(crate::sdt::Signature::FADT)?
+                .ok_or(AcpiError::TableMissing(crate::sdt::Signature::FADT))?
+        };
+
+        Ok(PmTimer { io_base: fadt.pm_timer_block, supports_32bit: fadt.flags.get_bit(8) })
```

Comment:
This needs to handle two more things:
- If `X_PM_TMR_BLK` is present (check with [`ExtendedField::access`](https://github.com/rust-osdev/acpi/blob/master/acpi/src/sdt.rs#L14)), use that instead of `PM_TMR_BLK`. Actually, this logic is probably best implemented as a `pm_timer_block()` method on `Fadt`, like we currently do with [`dsdt_address`](https://github.com/rust-osdev/acpi/blob/master/acpi/src/fadt.rs#L104)
- If both fields are zero, there is no PM timer. This should then return `Ok(None)`.